### 1. Даны две  независимые выборки. Не соблюдается условие нормальности
x1  380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции.


In [1]:
import numpy as np 
import scipy.stats as stats

In [2]:
x1 = [380, 420, 290]
y1 = [140, 360, 200, 900]
a = 0.05

In [6]:
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

H0 hypothesis is accepted, H1 is rejected because 0.629 > 0.05

### 2. Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

In [7]:
first = [150, 160, 165, 145, 155]
second = [140, 155, 150, 130, 135]
third = [130, 130, 120, 130, 125]
a = 0.05

In [8]:
stats.friedmanchisquare(first, second, third)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Ho is rejected and H1 is accepted because 0.008 < 0.05

### 3. Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [9]:
first = [150, 160, 165, 145, 155]
second = [140, 155, 150, 130, 135]
a = 0.05

In [10]:
stats.wilcoxon(first, second)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

H0 is rejected and H1 is accepted because 0.0625(p-value) > 0.05(alpha)

### 4. Даны 3 группы  учеников плавания.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54


In [11]:
g1 = [56, 60, 62, 55, 71, 67, 59, 58, 64, 67]
g2 = [57, 58, 69, 48, 72, 70, 68, 71, 50, 53]
g3 = [57, 67, 49, 48, 47, 55, 66, 51, 54]
a = 0.05

In [12]:
stats.kruskal(g1, g2, g3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

H0 is accepted and H1 is rejected because 0.065 > 0.05

### 5. Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%.

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [13]:
array = [2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]
n = 10
mu = 2.5

In [15]:
mean = np.mean(array) 
sigma = np.std(array, ddof = 1) 

In [16]:
tn = (mean - mu) / (sigma / n**0.5)
print(f't-value: {round(tn, 4)}')

t-value: 0.5631


In [17]:
a = 0.05

t1 = stats.t.ppf(1 - a / 2, n - 1)
t2 = stats.t.ppf(a / 2, n - 1)
print(f'Table t-value;: t2 = {round(t2, 4)}, t1 = {round(t1, 4)}')

Table t-value;: t2 = -2.2622, t1 = 2.2622
